frontend/src/components/containers/ContainersDetailContent.tsx

```diff
@@ -26,46 +31,37 @@ const tabStyle = {
   height: "100%",
 }
 
-const mapStateToProps = state => ({
-  containersByID: state.containers.itemsByID,
-  coordinatesByID: state.coordinates.itemsByID,
-  containerKindsByID: state.containerKinds.itemsByID,
-});
-
-const actionCreators = { get, listParents };
-
-const ContainersDetailContent = ({
-  containersByID,
-  coordinatesByID,
-  containerKindsByID,
-  get,
-  listParents
-}) => {
+const ContainersDetailContent = ({}) => {
   const history = useNavigate();
   const { id } = useParams();
   const [activeKey, setActiveKey] = useHashURL('overview')
 
-  const container = containersByID[id] || {};
-  // const error = container.error;
-  const isFetching = !containersByID[id] || container.isFetching;
-  const isLoaded = containersByID[id] && container.isLoaded;
-  let experimentRunsIDs = []
+  const dispatch = useAppDispatch()
 
-  if (!isLoaded)
-    get(id);
+  const containersByID = useAppSelector(selectContainersByID)
+  const coordinatesByID = useAppSelector(selectCoordinatesByID)
+  const containerKindsByID = useAppSelector(selectContainerKindsByID)
 
-  if (isLoaded && !container.parents)
-    listParents(id);
+  const container: Container | undefined = id && containersByID[id];
+  // const error = container.error;
+  const isFetching = container ? container.isFetching : true;
+  const isLoaded = container ? container.isLoaded : false;
+  const experimentRunsIDs = isLoaded && container?.experiment_run ? [container.experiment_run] : []
 
-  if (isLoaded)
-    if (container.experiment_run)
-      experimentRunsIDs.push(container.experiment_run)
+  useEffect(() => {
+    if (!isLoaded)
+      dispatch(get(id));
+  }, [isLoaded, dispatch, id])
 
+  useEffect(() => {
+    if (isLoaded && !container?.parents)
+      dispatch(listParents(id));
+  }, [isLoaded, container?.parents, dispatch, id])
 
   return (
     <>
       <AppPageHeader
-        title={`Container ${container.name || id}`}
+        title={`Container ${container?.name ?? id}`}
```

Comment:
The "||" would be safer option here since the "??" operator would take "" as a valid value and return it ... This is not possible normally with the container.name but then the id option would never be called on anything but null or undefined.